services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -106,31 +108,85 @@ async def list_datasets(self, user_id: UserID) -> list[DatasetMetaData]:
+            self.simcore_bucket_name,
+            prefix=ensure_ends_with(fmd.file_id, "/"),
+            max_files_to_list=max_items_to_include,
+        )
+        result: list[FileMetaData] = [
+            convert_db_to_model(
+                FileMetaDataAtDB(
+                    location_id=fmd.location_id,
+                    location=fmd.location,
+                    bucket_name=fmd.bucket_name,
+                    object_name=x.file_id,
+                    user_id=fmd.user_id,
+                    # NOTE: to ensure users have a consistent experience the
+                    # `created_at` field is inherited from the last_modified
+                    # coming from S3. This way if a file is created 1 month after the
+                    # creation of the directory, the file's creation date
+                    # will not be 1 month in the passed.
+                    created_at=x.last_modified,
+                    file_id=x.file_id,
+                    file_size=parse_obj_as(ByteSize, x.size),
+                    last_modified=x.last_modified,
+                    entity_tag=x.e_tag,
+                    is_soft_link=False,
+                    is_directory=False,
+                )
+            )
+            for x in files_in_folder
+        ]
+        return result
+
     async def list_files(
-        self, user_id: UserID, uuid_filter: str = ""
+        self, user_id: UserID, expand_dirs: bool, uuid_filter: str = ""
     ) -> list[FileMetaData]:
+        """
+        expand_dirs `False`: returns the one metadata entry for each directory
+        expand_dirs `True`: returns all files in each directory (no directories will be included)
+
+        NOTE: expand_dirs will be replaced by pagination in the future
+        currently only {EXPAND_DIR_MAX_ITEM_COUNT} items will be returned
+        The endpoint produces similar results to what it did previously
+        """
+
         data: list[FileMetaData] = []
-        accesible_projects_ids = []
+        accessible_projects_ids = []
         async with self.engine.acquire() as conn, conn.begin():
-            accesible_projects_ids = await get_readable_project_ids(conn, user_id)
-            file_metadatas: list[
+            accessible_projects_ids = await get_readable_project_ids(conn, user_id)
+            file_meta_data: list[
                 FileMetaDataAtDB
             ] = await db_file_meta_data.list_filter_with_partial_file_id(
                 conn,
                 user_id=user_id,
-                project_ids=accesible_projects_ids,
+                project_ids=accessible_projects_ids,
                 file_id_prefix=None,
                 partial_file_id=uuid_filter,
+                only_files=False,
             )
 
-            for fmd in file_metadatas:
+            # adding all the file_meta_data entries before attempting
+            for fmd in file_meta_data:
+                if fmd.is_directory and expand_dirs:
```

Comment:
why do you check is_directory here? that is only what you got from your query above. there are no files here. I think there is a problem in the logic no?